### ISSUE TITLE
Fix failing bamboo job with ITC hardware

### ISSUE THREAD

**Issue by t-b** [AllenInstitute/MIES#1498]
Since around February 2022 the ITC tests are mostly failing in CI. See http://bamboo.corp.alleninstitute.org/reports/generateReport.action?reportKey=com.atlassian.bamboo.plugin.system.reports%3AratioOfSuccess&selectFields=reportKey&labelTarget=&buildIds=153157784&selectFields=buildIds&groupByPeriod=AUTO&selectFields=groupByPeriod&dateFilter=RANGE&selectFields=dateFilter&dateFrom=01%2F01%2F2020&dateTo=31%2F12%2F2022&save=Submit&atl_token=20bfbb9788561001a2d14b27b31b40bc663b6fc7 (requires VPN access)

Graph:

![grafik](https://user-images.githubusercontent.com/1185677/193652451-c61ed2a6-967e-4c75-b2f4-5f8d67118614.png)

We would like to get that fixed as it is bad.

- We do get quite often a crash with ITC hardware in CI. We also have the PDB files on the CI server. This is tracked at WaveMetrics as issue 4168. The stacktrace looks like:

```
2022-05-02 09:30:38: BUG: Unexpected magic value in IgorStringClass::v().
Stack Trace:
	QApplicationPrivate::notify_helper
	QObject::event
	HandleXOPWinMessage
	HandleXOPWinMessage
	SetScrapData
	SetScrapData
	BlockMove
	BlockMove
	BlockMove
	BlockMove
	SetScrapData
	SetScrapData
	SetScrapData
	SetScrapData
	SetScrapData
	SetScrapData
	BlockMove
	BlockMove
	BlockMove
	BlockMove
	SetScrapData
	HandleXOPWinMessage
	HandleXOPWinMessage
```

Which does not make any sense at all as none of our XOPs use `SetScrapData`. This function does also not call itself.

We don't get these crashes with NI hardware.

- The timepoint in spring 2022 where the number of sucessfull builds drops does not coincide with any big things like IP8->IP9
- The ITC XOP was last updated in d1830b66 (XOPS-IP9-64bit/itcXOP2-64.xop: Update it, 2022-05-26).

**Comment by t-b** [comment]
I ran HardwareBasic.pxp overnight with ITC hardware. It crashed in ITCstartAcq2 via

![grafik](https://user-images.githubusercontent.com/1185677/193804281-0a8c6d46-cbce-4bfe-a125-87bbb33689a6.png)

Unfortunately did then the debugger also crash.

**Comment by t-b** [comment]
W10DTMJ365MB-svc-multipatch-2022-05-20T17_21_15Z\Crash 2022-05-19 16-00-38.dmp

Stacktrace from CDB is wrong. The one from VS 2022 looks much better.

Observations:
- Main thread
- Called from background function, either testcase _REENTRY or aquisition DQM_FIFOMonitor or similiar
- It is crashing when deleting a free text wave. Either in a free datafolder or wave ref wave
- It is not a TBB wave

Text wave has sizes of 9733 and 34831.

The invalid pointers have 16bit patterns e.g. 0x15ba15ba15ba15ba or 0xffcfffcfffcdffce. The ITCXOP2 uses 16bit signed integer wave (Make/W).

**Comment by t-b** [comment]
Passing builds from Feb 2022, http://bamboo.corp.alleninstitute.org/download/MIES-PLAYIN-TWIO/build_logs/MIES-PLAYIN-TWIO-635.log, are already using IP9.

**Comment by t-b** [comment]
I've ran HardwareTests.pxp with ITC hardware locally now with an IP9 debug version. It crashed with instrumentation and an "unexpectec magic value" error in the testcase RestoreDAEphysPanel. This test is with us since e39315e0 (Tests for Load/Save of GUIState and Experiment state through DAEphys panel, 2019-11-13).

I'm now running that test in a loop to see if that happens again easily.

**Comment by t-b** [comment]
I've now ran RestoreDAEphysPanel ten times in a row with instrumentation. No crash.

**Comment by t-b** [comment]
Tonight no crash but it failed at

```
 Entering test case "HandlesStuckFIFOProperly:ITC"
  !!! Assertion FAILED !!!
  Message: "Writing scaledDataWave failed, please save the experiment and file a bug report: fifoPosGlobal 104583, fifoPos 12345, scaledDataWave rows 104583, stopCollectionPoint 104583"
```

**Comment by t-b** [comment]
Two more crashes: One with HardwareAnalysisFunctions.pxp (unexpected magic) and one deep inside QT (HardwareBasic.pxp).

Otherwise I had 5 successful runs without crashes with HardwareAnalysisFunctions.pxp

**Comment by t-b** [comment]
One more crash with expensive and instrumentation:

```
00000000`00bf9e90 00000000`159c9b2e itcXOP2_64!AcqCallBack+0x59f
00000000`00bf9f20 00000000`1599f58d itcXOP2_64!ITC_Start+0x38e
00000000`00bf9f90 00000000`159ab39f itcXOP2_64!ITCDLL::ITC_Start(class DeviceIDHelper * DeviceID = 0x00000000`00bfa088, struct ITCStartInfo * sParam = 0x00000000`00000000)+0x1bd [C:\gitlab-runner\builds\VSme1h_7\0\mirror\itcxop2\src\itcdll.cpp @ 680]
00000000`00bfa050 00007ff7`b8cd254c itcXOP2_64!ExecuteITCStartAcq2(struct ITCStartAcq2RuntimeParams * p = 0x00000000`1e20ac44)+0x16f [C:\gitlab-runner\builds\VSme1h_7\0\mirror\itcxop2\src\ITCStartAcq2.cpp @ 225]
00000000`00bfa0f0 00007ff7`b76acfa1 Igor64!CallXOP(struct IORec ** ioRecHandle = 0x00000000`1c003e90, <function> * XOPRoutine = 0x00000000`159ab230, void * XOPParam = 0x00000000`1e20ac44)+0x2c [e:\projekte\wavemetrics\igor9\igorsrc\windows\xopwin.cpp @ 1033]
00000000`00bfa130 00007ff7`b7400a04 Igor64!ExecuteXOPOperation(struct IORec ** ioRecHandle = 0x00000000`1c003e90, char * operationName = 0x00000000`1e20a9b0 "ITCStartAcq2", void * runtimeParamsStartPtr = 0x00000000`1e20ac44, <function> * runtimeAddress = 0x00000000`159ab230)+0x61 [e:\projekte\wavemetrics\igor9\igorsrc\igor3\xopoperations.cpp @ 269]
00000000`00bfa170 00007ff7`b72bea51 Igor64!RunXOPOperation(int64 * pp = 0x00000000`10c931e8)+0x194 [e:\projekte\wavemetrics\igor9\igorsrc\igor2\operationhandler.cpp @ 8855]
00000000`00bfa1f0 00007ff7`b731d306 Igor64!C_RunCommand(void)+0x51 [e:\projekte\wavemetrics\igor9\igorsrc\igor2\compilerp2.cpp @ 17453]
00000000`00bfa230 00007ff7`b73037a1 Igor64!CExecuteSubrGeneral(int64 * newPC = 0x00000000`00bfa2c0)+0xc6 [e:\projekte\wavemetrics\igor9\igorsrc\igor2\functions.cpp @ 2552]
00000000`00bfa290 00007ff7`b73039da Igor64!ExecuteWMBIStructFunc(void * p = 0x00000000`2d23b720, int ufIndex = 0n34856)+0x151 [e:\projekte\wavemetrics\igor9\igorsrc\igor2\functions.cpp @ 3195]
00000000`00bfa2f0 00007ff7`b6dc342e Igor64!ExecuteWMBIStructFuncEx(void * p = 0x00000000`2d23b720, int ufIndex = 0n34856, int imIndex = 0n0)+0x4a [e:\projekte\wavemetrics\igor9\igorsrc\igor2\functions.cpp @ 3275]
00000000`00bfa330 00007ff7`b6dc1b02 Igor64!ExecuteNamedBackgroundTasks(void)+0x5de [e:\projekte\wavemetrics\igor9\igorsrc\igor1\background.cpp @ 894]
00000000`00bfa530 00007ff7`b5a99b4a Igor64!DoBackground(void)+0x32 [e:\projekte\wavemetrics\igor9\igorsrc\igor1\background.cpp @ 203]
00000000`00bfa570 00007ff7`b5a9a662 Igor64!IgorAppObject::doIdleEvent(bool allowReentry = false)+0x21a [e:\projekte\wavemetrics\igor9\igorsrc\src\igorapplication\igorappobject.cpp @ 4849]
00000000`00bfa610 00007ff9`9db5ba53 Igor64!IgorAppObject::timerEvent(class QTimerEvent * event = 0x00000000`00bfc070)+0x32 [e:\projekte\wavemetrics\igor9\igorsrc\src\igorapplication\igorappobject.cpp @ 4240]
00000000`00bfa660 00007ff9`9dafed55 Qt5Cored!QObject::event(class QEvent * e = 0x00000000`00bfc070)+0x93 [c:\qtbuild512\qt\qtbase\src\corelib\kernel\qobject.cpp @ 1243]
00000000`00bfaec0 00007ff9`9e563ae5 Qt5Cored!QCoreApplication::event(class QEvent * e = 0x00000000`00bfc070)+0x35 [c:\qtbuild512\qt\qtbase\src\corelib\kernel\qcoreapplication.cpp @ 1958]
00000000`00bfaef0 00007ff9`9f394280 Qt5Guid!QGuiApplication::event(class QEvent * e = 0x00000000`00bfc070)+0x55 [c:\qtbuild512\qt\qtbase\src\gui\kernel\qguiapplication.cpp @ 1863]
00000000`00bfaf30 00007ff7`b5a9a616 Qt5Widgetsd!QApplication::event(class QEvent * e = 0x00000000`00bfc070)+0x590 [c:\qtbuild512\qt\qtbase\src\widgets\kernel\qapplication.cpp @ 1955]
00000000`00bfb060 00007ff9`9f396102 Igor64!IgorAppObject::event(class QEvent * event = 0x00000000`00bfc070)+0x1f6 [e:\projekte\wavemetrics\igor9\igorsrc\src\igorapplication\igorappobject.cpp @ 4229]
00000000`00bfb0e0 00007ff9`9f3903a8 Qt5Widgetsd!QApplicationPrivate::notify_helper(class QObject * receiver = 0x00000000`00dc2830, class QEvent * e = 0x00000000`00bfc070)+0x1a2 [c:\qtbuild512\qt\qtbase\src\widgets\kernel\qapplication.cpp @ 3650]
00000000`00bfb140 00007ff7`b5a95948 Qt5Widgetsd!QApplication::notify(class QObject * receiver = 0x00000000`00dc2830, class QEvent * e = 0x00000000`00bfc070)+0x528 [c:\qtbuild512\qt\qtbase\src\widgets\kernel\qapplication.cpp @ 3006]
00000000`00bfbe30 00007ff9`9daff2e0 Igor64!IgorAppObject::notify(class QObject * object = 0x00000000`00dc2830, class QEvent * event = 0x00000000`00bfc070)+0x268 [e:\projekte\wavemetrics\igor9\igorsrc\src\igorapplication\igorappobject.cpp @ 968]
00000000`00bfbf80 00007ff9`9dafcea2 Qt5Cored!QCoreApplication::notifyInternal2(class QObject * receiver = 0x00000000`00dc2830, class QEvent * event = 0x00000000`00bfc070)+0x120 [c:\qtbuild512\qt\qtbase\src\corelib\kernel\qcoreapplication.cpp @ 1088]
00000000`00bfc000 00007ff9`9dbcd8c4 Qt5Cored!QCoreApplication::sendEvent(class QObject * receiver = 0x00000000`00dc2830, class QEvent * event = 0x00000000`00bfc070)+0x42 [c:\qtbuild512\qt\qtbase\src\corelib\kernel\qcoreapplication.cpp @ 1477]
00000000`00bfc030 00007ff9`9dbca972 Qt5Cored!QEventDispatcherWin32Private::sendTimerEvent(int timerId = 0n309)+0x94 [c:\qtbuild512\qt\qtbase\src\corelib\kernel\qeventdispatcher_win.cpp @ 458]
00000000`00bfc0a0 00007ffa`1f49e858 Qt5Cored!qt_internal_proc(struct HWND__ * hwnd = 0x00000000`01b50b92 0+0+136x39 'QEventDispatcherWin32_Internal_Widget140710064254189', unsigned int message = 0x113, unsigned int64 wp = 0x135, int64 lp = 0n0)+0x672 [c:\qtbuild512\qt\qtbase\src\corelib\kernel\qeventdispatcher_win.cpp @ 240]
```

**Comment by t-b** [comment]
We think and hope this is now fixed with #1703.
